src/components/layout/PostLoginLayout.tsx

```diff
@@ -0,0 +1,64 @@
+import { useEffect } from 'react';
+import { Outlet, useNavigate } from 'react-router-dom';
+
+import styled from 'styled-components';
+
+import useCheckAccessToken from '../../hooks/useCheckAccessToken';
+
+import checkUserType from '../../utils/checkUserType';
+
+import Header from './Header';
+
+export default function PostLoginLayout() {
+  const navigate = useNavigate();
+
+  const validUserType = checkUserType();
+
+  const ready = useCheckAccessToken();
+
+  useEffect(() => {
+    if (!validUserType) {
```

Comment:
p3: validUserType이라는 변수에 결과값을 따로 담기보다는 아래처럼 바로 함수를 사용해도 좋을 것 같아요. 그리고 함수명을 보면 유저의 타입을 체크하는 것 같은데 실제로는 유저의 타입이 valid한지를 알려주고 있어서 미세하게 다르다고 생각하는데요. 함수명이 조금 더 자세하게 작성되어도 좋을 것 같아요

```suggestion
  useEffect(() => {
    if (!checkUserType()) {
```